applications/IgaApplication/custom_elements/iga_truss_element.cpp

```diff
@@ -0,0 +1,182 @@
+/*
+//  KRATOS  _____________
+//         /  _/ ____/   |
+//         / // / __/ /| |
+//       _/ // /_/ / ___ |
+//      /___/\____/_/  |_| Application
+//
+//  Main authors:   Anna Bauer
+//                  Thomas Oberbichler
+//                  Tobias Teschemacher
+*/
+
+// System includes
+#include "includes/define.h"
+#include "includes/variables.h"
+
+// External includes
+
+// Project includes
+#include "iga_truss_element.h"
+#include "iga_application_variables.h"
+
+namespace Kratos {
+
+Element::Pointer IgaTrussElement::Create(
+    IndexType NewId,
+    NodesArrayType const& ThisNodes,
+    PropertiesType::Pointer pProperties) const
+{
+    auto geometry = GetGeometry().Create(ThisNodes);
+
+    return Kratos::make_shared<IgaTrussElement>(NewId, geometry,
+        pProperties);
+}
+
+void IgaTrussElement::GetDofList(
+    DofsVectorType& rElementalDofList,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    rElementalDofList.resize(NumberOfDofs());
+
+    for (std::size_t i = 0; i < NumberOfNodes(); i++) {
+        SetDof(rElementalDofList, i, 0, DISPLACEMENT_X);
+        SetDof(rElementalDofList, i, 1, DISPLACEMENT_Y);
+        SetDof(rElementalDofList, i, 2, DISPLACEMENT_Z);
+    }
+
+    KRATOS_CATCH("")
+}
+
+void IgaTrussElement::EquationIdVector(
+    EquationIdVectorType& rResult,
+    ProcessInfo& rCurrentProcessInfo)
+{
+    KRATOS_TRY;
+
+    rResult.resize(NumberOfDofs());
+
+    for (std::size_t i = 0; i < NumberOfNodes(); i++) {
+        SetEquationId(rResult, i, 0, DISPLACEMENT_X);
```

Comment:
somehow my comment got lost...
here we go again:

I really think that you should not use your own function here for the following reasons:
- It makes the code harder to read, since you don't follow the standard procedure I assume you do sth special, which you are not
- The performance of this is not the best => take a look at the fluid-elements for how this should be done
- Every derived function has a custom implementation of this entire function anyway
- `SetEquationId` is a function of `Dof` and called by the `BuilderAndSolver`, this is also confusing here ...